source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -67,38 +68,40 @@ void generateV2Header(const std::string& src_addr, const std::string& dst_addr,
     addr_length[1] = PROXY_PROTO_V2_ADDR_LEN_INET;
     out.add(addr_length, 2);
 
-    uint8_t addrs[8];
+    MemBlockBuilder<uint8_t> addrs(8);
```

Comment:
Conceptually I like what you did here but it does add a dynamic memory allocation. Also I think this will leak.

Maybe this justifies the new constructor to MemBlockBuilder from an earlier revision, so you can still just use the stack-allocated uint8 buffer.

Reply:
I was thinking the same 👍 . The errors from the ci pipeline comes from the AddressSanitizer, hence it must be related with the usage of dinamically allocated memory. I'll try to test it locally on other machine, since the one I am using at the moment takes hours (ages) to compile and run tests.